godot-core/src/builtin/vector3.rs

```diff
@@ -85,6 +87,215 @@ impl Vector3 {
+        let x = cubic_interpolate(self.x, b.x, pre_a.x, post_b.x, weight);
+        let y = cubic_interpolate(self.y, b.y, pre_a.y, post_b.y, weight);
+        let z = cubic_interpolate(self.z, b.z, pre_a.z, post_b.z, weight);
+
+        Self::new(x, y, z)
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn cubic_interpolate_in_time(
+        self,
+        b: Self,
+        pre_a: Self,
+        post_b: Self,
+        weight: f32,
+        b_t: f32,
+        pre_a_t: f32,
+        post_b_t: f32,
+    ) -> Self {
+        let x = cubic_interpolate_in_time(
+            self.x, b.x, pre_a.x, post_b.x, weight, b_t, pre_a_t, post_b_t,
+        );
+        let y = cubic_interpolate_in_time(
+            self.y, b.y, pre_a.y, post_b.y, weight, b_t, pre_a_t, post_b_t,
+        );
+        let z = cubic_interpolate_in_time(
+            self.z, b.z, pre_a.z, post_b.z, weight, b_t, pre_a_t, post_b_t,
+        );
+
+        Self::new(x, y, z)
+    }
+
+    pub fn direction_to(self, to: Self) -> Self {
+        (to - self).normalized()
+    }
+
+    pub fn distance_squared_to(self, to: Self) -> f32 {
+        self.to_glam().distance_squared(to.to_glam())
+    }
+
+    pub fn distance_to(self, to: Self) -> f32 {
+        self.to_glam().distance(to.to_glam())
+    }
+
+    pub fn dot(self, with: Self) -> f32 {
+        self.to_glam().dot(with.to_glam())
+    }
+
+    pub fn floor(self) -> Self {
+        Self::from_glam(self.to_glam().floor())
+    }
+
+    pub fn inverse(self) -> Self {
+        Self::new(1.0 / self.x, 1.0 / self.y, 1.0 / self.z)
+    }
+
+    pub fn is_equal_approx(self, to: Self) -> bool {
+        is_equal_approx(self.x, to.x)
+            && is_equal_approx(self.y, to.y)
+            && is_equal_approx(self.z, to.z)
+    }
+
+    pub fn is_finite(self) -> bool {
+        self.to_glam().is_finite()
+    }
+
+    pub fn is_normalized(self) -> bool {
+        self.to_glam().is_normalized()
+    }
+
+    pub fn is_zero_approx(self) -> bool {
+        is_zero_approx(self.x) && is_zero_approx(self.y) && is_zero_approx(self.z)
+    }
+
+    pub fn length_squared(self) -> f32 {
+        self.to_glam().length_squared()
+    }
+
+    pub fn lerp(self, to: Self, weight: f32) -> Self {
+        Self::from_glam(self.to_glam().lerp(to.to_glam(), weight))
+    }
+
+    pub fn limit_length(self, length: Option<f32>) -> Self {
+        Self::from_glam(self.to_glam().clamp_length_max(length.unwrap_or(1.0)))
+    }
+
+    pub fn max_axis_index(self) -> i32 {
+        let me = self.to_glam().max_element();
+        if me == self.x {
+            0
+        } else if me == self.y {
+            1
+        } else {
+            2
+        }
+    }
+
+    pub fn min_axis_index(self) -> i32 {
+        let me = self.to_glam().min_element();
+        if me == self.x {
+            0
+        } else if me == self.y {
+            1
+        } else {
+            2
+        }
+    }
```

Comment:
I'm not sure we should implement it like that (first computing the maximum, then comparing it to each axis).
Maybe check [the `gdnative` implementation](https://github.com/godot-rust/gdnative/blob/master/gdnative-core/src/core_types/vector3.rs#L218-L258). 

We should also make use of the `Vector3Axis` type, defined further below in this file.

Reply:
copy, fixing to use the given type